subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvoker.java

```diff
@@ -392,6 +392,26 @@ public void visitOutputTrees(CacheableTreeVisitor visitor) {
             throw new UnsupportedOperationException("we don't cache yet");
         }
 
+        @Override
+        public Optional<AfterPreviousExecutionState> getAfterPreviousExecutionState() {
+            return Optional.empty();
+        }
+
+        @Override
+        public boolean deleteOutputsBeforeExecution() {
+            return false;
```

Comment:
It would be nice to be able to delete the outputs before the transformation executes by using this flag. The only thing missing is to register the global transforms directory as safe to delete right?